### PR TITLE
Zone 5 -CrashTestDummies

### PR DESCRIPTION
**Pull request by TomBrunner**
User Story:
As a programmer I want to have this percept 
So that I can use details from the zone.
#9

new classes:
J2Zone
J2ZoneTest

### CODE REVIEW

**Comment by HarmenKroon** [comment]
As discussed in #9 zone/6 has been changed to zone/5.
This is how the percepts is implemented:
`zone/5 - zone(<zone_ID>,<name>,<size>,<max_floors>, <allowed_catagories>)`

`<zone_ID>` - ID given to the zone
`<name>` - Name of the zone
`<max_floor>` - The maximum allowed amount of floors to be set for the zone
`<size>` -  The Amount of squared meters of zone ground
`<allowed_catagories>` - The allowed catagories of the zone

File: environment/src/test/java/tygronenv/translators/J2ZoneTest.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+	@Test
+	public void testTranslate() throws TranslationException {
+		translator.translate(z);
+		verify(z, atLeast(1)).getMaxAllowedFloors();
+		verify(z, atLeast(1)).getMultiPolygon();
+		verify(z, atLeast(1)).getAllowedCategories();
+	}
+
+	/* // OLD
+	 * package tygronenv.translators;
```
Comment (levilime) [inline environment/src/test/java/tygronenv/translators/J2ZoneTest.java]: Is there a reason for this commented test code in your PR? If it won't be run then please delete it.

**Comment by levilime** [comment]
Please wait with the merge of this ticket untill #33 is merged, so from sprint week 5 onward all PR's are checked with checkstyle, cobertura (CI integrated) and findbugs, PMD (insight through output log on CI)

**Comment by MaxGroenenboom** [comment]
Why is the commented test code @levilime mentioned in there? Commented code should not be in the pull request. Apart from that it looks good.

**Comment by spiderbiggen** [comment]
Same as said above. Looks good, but why the commented test. 
- Rampestampers

**Comment by TomBrunner** [comment]
I had forgotten about those, the comments have been removed now.

**Comment by levilime** [comment]
can you add the link of the travis build for this commit  e3ad446

**Comment by levilime** [comment]
It seems your build is failing: https://travis-ci.org/nskartoredjo/tygron/builds/132480775

**Comment by jannelie** [comment]
https://travis-ci.org/nskartoredjo/tygron/builds/132517258
We fixed the build

**Comment by MaxGroenenboom** [comment]
I see there still is a minor edit in the readme (only a space), and we shouldn't touch that file. I do approve of the rest of the PR though.

**Comment by TomBrunner** [comment]
it's fixed

**Comment by MaxGroenenboom** [comment]
Looks perfect now ;)

**Comment by klikooo** [comment]
Does it work in GOAL? Cause there is an exception in the travis buid log:

eis.eis2java.exception.NoTranslatorException: No translator found for class nl.tytech.data.engine.item.Zone
    at eis.eis2java.translation.Translator.translate2Parameter(Translator.java:128)
    at eis.eis2java.translation.Translator$CollectionTranslator.translate(Translator.java:351)
    at eis.eis2java.translation.Translator$CollectionTranslator.translate(Translator.java:339)
    at eis.eis2java.translation.Translator.translate2Parameter(Translator.java:132)
    at tygronenv.EntityEventHandler.createPercepts(EntityEventHandler.java:139)
    at tygronenv.EntityEventHandler.notifyListener1(EntityEventHandler.java:111)
    at tygronenv.EntityEventHandler.notifyListener(EntityEventHandler.java:81)
    at nl.tytech.core.client.event.EventManager.lambda$fireToNormalEvent$9(EventManager.java:796)
    at nl.tytech.core.client.event.EventManager$$Lambda$34/1393992702.run(Unknown Source)
    at nl.tytech.core.client.concurrent.SliceManager._exec(SliceManager.java:100)
    at nl.tytech.core.client.concurrent.SliceManager.exec(SliceManager.java:52)
    at nl.tytech.core.client.event.EventManager.threadPusher(EventManager.java:297)
    at nl.tytech.core.client.event.EventManager.fireToNormalEvent(EventManager.java:794)
    at nl.tytech.core.client.event.EventManager.fireEvent(EventManager.java:707)
    at nl.tytech.core.client.event.EventManager.fire(EventManager.java:156)
    at nl.tytech.core.client.net.Status.fireListUpdateEvent(Status.java:156)
    at nl.tytech.core.client.net.Status.lambda$updateVersions$10(Status.java:395)
    at nl.tytech.core.client.net.Status$$Lambda$37/456140082.run(Unknown Source)
    at nl.tytech.core.client.concurrent.SliceManager$1.run(SliceManager.java:152)
    at nl.tytech.core.client.concurrent.SliceManager.updateParallel(SliceManager.java:182)
    at nl.tytech.core.client.concurrent.UpdateManager.updateParallel(UpdateManager.java:217)
    at nl.tytech.core.client.concurrent.UpdateManager.access$200(UpdateManager.java:19)
    at nl.tytech.core.client.concurrent.UpdateManager$ParallelUpdater.update(UpdateManager.java:83)
    at nl.tytech.core.client.concurrent.LimitedFPSThread.run(LimitedFPSThread.java:56)

In EisEnv.java you can add the translator for the zone.

**Comment by TomBrunner** [comment]
Ik had niet meegekregen dat je eis.env.java ook moet updaten, heb het toegevoegt.

De enige exception die ik momenteel nog krijg is op regel 4085: eis.exceptions.EntityException: Entity "Municipality" does already exist

**Comment by jannelie** [comment]
https://travis-ci.org/nskartoredjo/tygron/builds/132793860 Finally passed, can somebody merge?

**Comment by spiderbiggen** [comment]
looks good / RAMpestampers

**Comment by klikooo** [comment]
Approved / WhySoSerious

**Comment by MHMCD** [comment]
looks good 

\- CodeFox [CodeFox](https://github.com/codefoxnl)

**Pull request merged**